### PR TITLE
Use Fillout form for feedback with prefilled debug data

### PR DESCRIPTION
**Pull request by YodaLightsabr**
## Summary of the problem
#7456 is great! We don't have debug data for bug reports, though.

## Describe your changes
- Integrates the Fillout form with the support button (which gives routing data & user information prefilled)
- Makes the support button have a pointer cursor
- Makes "Contact Us" title case
- Removes old `static_pages#feedback` method

image

https://forms.hackclub.com/hcb-feedback

Closes #7540

### CODE REVIEW

File: app/views/application/_support_button.html.erb [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
     </a>
     <% if current_user&.events&.not_demo_mode&.any? %>
-       <%= link_to "https://hack.af/bank-userfeedback?prefill_Name=#{current_user&.name}&prefill_Email=#{current_user&.email}&prefill_Organization=#{current_user&.events&.first&.name}", class: "support-menu__item", target: "_blank" do %>
+      <% routing = Rails.application.routes.recognize_path(params[:page_path]) %>
+
+      <% url_params = {
+           name: current_user&.name,
+           email: current_user&.email,
+           url: "#{request.base_url}#{params[:page_path]}",
+           location: "#{routing[:controller]}##{routing[:action]} #{routing[:id] if routing[:id] && share_email}".strip,
```
Comment (garyhtou) [inline app/views/application/_support_button.html.erb]: we can drop the `share_email`

**Comment by garyhtou** [comment]
Feedback on the form:

Add a radio box for "Support". Have it go to an end screen saying email hcb@

File: app/views/application/_support_button.html.erb (hunk 1 above)
Comment (YodaLightsabr) [reply app/views/application/_support_button.html.erb]: Good catch

**Review by garyhtou — COMMENTED** [review]
🪄 great! give it a test (if you haven't already), and merge!

File: app/views/application/_support_button.html.erb [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+      <%= inline_icon "email", size: 24 %> Contact Us
     </a>
     <% if current_user&.events&.not_demo_mode&.any? %>
-       <%= link_to "https://hack.af/bank-userfeedback?prefill_Name=#{current_user&.name}&prefill_Email=#{current_user&.email}&prefill_Organization=#{current_user&.events&.first&.name}", class: "support-menu__item", target: "_blank" do %>
+      <% routing = Rails.application.routes.recognize_path(params[:page_path]) %>
+
+      <% url_params = {
+           name: current_user&.name,
+           email: current_user&.email,
+           url: "#{request.base_url}#{params[:page_path]}",
```
Comment (garyhtou) [inline app/views/application/_support_button.html.erb]: is `params[:page_path]` set anywhere?

Comment (garyhtou) [inline app/views/application/_support_button.html.erb]:
```suggestion
      <% routing = Rails.application.routes.recognize_path(request.original_fullpath) %>

      <% url_params = {
           name: current_user&.name,
           email: current_user&.email,
           url: "#{request.original_fullpath}",
```

**Review by manuthecoder — APPROVED** [review]
Looks great to me! 🎉

**Review by sampoder — APPROVED** [review]
big buttons!

**Pull request merged**